sway-core/src/asm_generation/asm_builder.rs

```diff
@@ -226,11 +226,16 @@ impl<'ir> AsmBuilder<'ir> {
                     }
                 }
                 Instruction::Revert(revert_val) => self.compile_revert(instr_val, revert_val),
-                Instruction::StateLoadQuadWord { load_val, key } => check!(
+                Instruction::StateLoadQuadWord {
+                    load_val,
+                    key,
+                    number_of_slots,
+                } => check!(
```

Comment:
we should also have `previously_set` based on `$rB`

Reply:
So I've made the _output_ of the IR instruction be the `previous_set` flag, as in:
```
previously_set = state_load_quad(load_va, key, number_of_slots)
```
That way I don't have to worry about the value being passed in by ref to the instructions.